File: BE/WebParent/BE_Client/src/main/java/com/pbl5/client/service/StoreRequestService.java
package com.pbl5.client.service;

import com.pbl5.client.dto.StoreRequestDto;
import com.pbl5.common.entity.StoreRequest;

import java.util.List;

public interface StoreRequestService {

    public void save(StoreRequestDto request);

    List<StoreRequestDto> getAllStoreRequests();


    StoreRequest getStoreRequestByCustomerId(Integer customerId);
}

File: FE/FE_Client/src/services/store_request_service.ts
import axios from "../axios.customize";
import { StoreRequestDto, StoreRequestDtoImpl } from "../models/dto/StoreRequestDto";


class StoreRequestService {
    async save(storeRequest: StoreRequestDto): Promise<void> {
        try {
            const response = await axios.post("/store-requests/save", storeRequest);
            if (response.status === 201) {
                console.log("Store request saved successfully.");
            } else {
                console.error("Failed to save store request:", response.statusText);
            }
        }
        catch (error) {
            console.error("Error submitting review:", error);
        }
    }

    async getByCustomerId(customerId: number): Promise<StoreRequestDtoImpl | undefined> {
        try {
            const response = await axios.get<StoreRequestDtoImpl>(`/store-requests/customer/${customerId}`);
            return response.data;
        } catch (error) {
            console.error("Error fetching store requests:", error);
            return undefined;
        }
    }

}


export default StoreRequestService;